### PR TITLE
Fix primitive numeric types cast not to panic

### PR DESCRIPTION
**Pull request by earney**
Modify TryInto functions so that tests like the ones below do not panic but throw an ImpossibleCast Error.

```
test!(Value::I64(128), Err(ValueError::ImpossibleCast.into()));
test!(Value::F64(128.0), Err(ValueError::ImpossibleCast.into()));
```

### CODE REVIEW

**Comment by earney** [comment]
I realize that an any i8 will cast to an i64 , but figured it was best to use to_i64(), instead of as i64.  the compiler should be smart enough to realize this and can optimize the to_i64 away if it needs to.

File: core/src/data/value/into.rs [hunk 1]
```diff
@@ -534,6 +545,7 @@ mod tests {
         test!(Value::Bool(true), Ok(1.0));
         test!(Value::Bool(false), Ok(0.0));
         test!(Value::I8(122), Ok(122.0));
+        test!(Value::I64(122), Ok(122.0));
```
Comment (ever0de) [inline core/src/data/value/into.rs:548]:
```rust
        test!(Value::I64(122), Ok(122.0));
        test!(Value::I64(1234567890), Ok(1234567890.0));
```

If it is the same case, it seems that there is no need to add it.

File: core/src/data/value/into.rs [hunk 2]
```diff
@@ -417,6 +417,10 @@ mod tests {
             Err(ValueError::ImpossibleCast.into())
         );
         test!(Value::Null, Err(ValueError::ImpossibleCast.into()));
+
+        //impossible casts to bool
```
Comment (ever0de) [inline core/src/data/value/into.rs:421]: <https://doc.rust-lang.org/book/ch03-04-comments.html>
Similarly, in `GlueSQL` it is common to use a leading space like `// TEMP`.

Comment (ever0de) [inline core/src/data/value/into.rs:421]: <https://doc.rust-lang.org/book/ch03-04-comments.html>
Similarly, in `GlueSQL` it is common to use a leading space like `// TEMP`.

File: core/src/data/value/into.rs [hunk 3]
```diff
@@ -436,6 +440,7 @@ mod tests {
         test!(Value::I8(122), Ok(122));
         test!(Value::I64(122), Ok(122));
         test!(Value::F64(122.0), Ok(122));
+        test!(Value::F64(122.1), Ok(122));
```
Comment (ever0de) [inline core/src/data/value/into.rs:443]:
```rust
        test!(Value::F64(122.0), Ok(122));
        test!(Value::F64(122.1), Ok(122));
```

If it is the same case, it seems that there is no need to add it.

Comment (ever0de) [inline core/src/data/value/into.rs:443]:
```rust
        test!(Value::F64(122.0), Ok(122));
        test!(Value::F64(122.1), Ok(122));
```

If it is the same case, it seems that there is no need to add it.

File: core/src/data/value/into.rs [hunk 4]
```diff
@@ -484,8 +493,10 @@ mod tests {
         test!(Value::Bool(true), Ok(1));
         test!(Value::Bool(false), Ok(0));
         test!(Value::I8(122), Ok(122));
+        test!(Value::I64(122), Ok(122));
         test!(Value::I64(1234567890), Ok(1234567890));
         test!(Value::F64(1234567890.0), Ok(1234567890));
+        test!(Value::F64(1234567890.1), Ok(1234567890));
```
Comment (ever0de) [inline core/src/data/value/into.rs:499]:
```rust
        test!(Value::F64(1234567890.0), Ok(1234567890));
        test!(Value::F64(1234567890.1), Ok(1234567890));
```

If it is the same case, it seems that there is no need to add it.

Comment (ever0de) [inline core/src/data/value/into.rs:499]:
```rust
        test!(Value::F64(1234567890.0), Ok(1234567890));
        test!(Value::F64(1234567890.1), Ok(1234567890));
```

If it is the same case, it seems that there is no need to add it.

**Review by ever0de — COMMENTED** [review]
Can you consider deleting duplicate test cases? 🙏 
Thanks 👍

File: core/src/data/value/into.rs (hunk 2 above)
Comment (earney) [reply core/src/data/value/into.rs:421]: I did not know that. I will try to keep that in mind for future cases.

File: core/src/data/value/into.rs (hunk 3 above)
Comment (earney) [reply core/src/data/value/into.rs:443]: I will change to test from x.1 to x.9.  that example will check for round vs truncate.

**Review by ever0de — APPROVED** [review]
LGTM

File: core/src/data/value/into.rs [hunk 5]
```diff
@@ -152,9 +152,9 @@ impl TryInto<i64> for &Value {
                     0
                 }
             }
-            Value::I8(value) => *value as i64,
+            Value::I8(value) => value.to_i64().ok_or(ValueError::ImpossibleCast)?,
```
Comment (panarch) [inline core/src/data/value/into.rs:155]: This looks unnecessary (cast from smaller type to bigger)

File: core/src/data/value/into.rs [hunk 6]
```diff
@@ -191,8 +191,8 @@ impl TryInto<f64> for &Value {
                     0.0
                 }
             }
-            Value::I8(value) => *value as f64,
-            Value::I64(value) => (*value as f64).trunc(),
+            Value::I8(value) => value.to_f64().ok_or(ValueError::ImpossibleCast)?,
```
Comment (panarch) [inline core/src/data/value/into.rs:194]: This, too.

**Review by panarch — CHANGES_REQUESTED** [review]
Thanks a lot!
There are very tiny comments about cast from i8 to i64

**Review by panarch — APPROVED** [review]
All looks great, thanks a lot!

**Pull request merged**